src/neo4j_genai/generation/prompts.py

```diff
@@ -39,6 +42,10 @@ def __init__(
         self.template = template or self.DEFAULT_TEMPLATE
         self.expected_inputs = expected_inputs or self.EXPECTED_INPUTS
 
+        assert (
```

Comment:
Since we have the `expected_inputs`, can we use this list instead of a hard coded `query_text`? I'm thinking about something like this:

```
for e in self.expected_inputs:
    if f"{{{e}}}" not in self.template:
        raise SomeError
```

Reply:
We could create a `PromptMissingPlaceholderError` and run this check upon construction? I don't think we should expect users to provide a list of expected inputs though, so for custom prompts I think we can only check for the `query_text` placeholder.